src/script/api/script_rail.hpp

```diff
@@ -60,7 +60,7 @@ class ScriptRail : public ScriptObject {
 		RAILTRACK_SW_SE   = ::TRACK_BIT_LOWER,   ///< Track in the lower corner of the tile (south).
 		RAILTRACK_NW_SW   = ::TRACK_BIT_LEFT,    ///< Track in the left corner of the tile (west).
 		RAILTRACK_NE_SE   = ::TRACK_BIT_RIGHT,   ///< Track in the right corner of the tile (east).
-		RAILTRACK_INVALID = ::INVALID_TRACK_BIT, ///< Flag for an invalid track.
+		RAILTRACK_INVALID = 0xFF, ///< Flag for an invalid track.
```

Comment:
It's a bit cross purposes because it comes from a different enum, but RAILTRACK_INVALID seems to be used like INVALID_TRACK, perhaps it should be defined in the same way?

Reply:
```
src/script/api/script_rail.cpp: if (!IsRailStationTile(tile)) return RAILTRACK_INVALID;
src/script/api/script_rail.cpp: if (!IsRailTile(tile)) return RAILTRACK_INVALID;
```

Looks like "invalid track" to me

Reply:
I'm just saying INVALID_TRACK _could_ be used instead of a raw number

Reply:
Yes, in the same way I could use `INVALID_TRACKDIR`, `INVALID_DIR`, `INVALID_AXIS`. They are all `0xFF`.

In my opinion mixing enumeration types that are tangentially related in this `enum` will only bring more questions later on.

It will also work against @PeterN's work of splitting enums and enum-bitsets, as `TrackBits` is essentially `EnumBitSet<Track>`.

And yes, the places where `RAILTRACK_INVALID` gets returned can look like they are returning a track at a quick glance, especially the station one, but those functions are returning essentially `TrackBits`.

Finally, `RAILTRACK_INVALID` is something specific to the Script API. I had to leave it in because it's used there, and it needs to retain its original value to prevent issues with script compatibility.